File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/MedicationStatement.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class MedicationStatement : DomainResource, IResource
{
    public Base.CodeableConcept? Category { get; set; }
    public string[]? Dosage { get; set; }
    public Base.ResourceReference[]? DerivedFrom { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public Base.CodeableConcept? MedicationCodeableConcept { get; set; }
    public Base.CodeableConcept[]? StatusReason { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public string? EffectiveDateTime { get; set; }
    public required string Status { get; set; }
    public object? Effective    
    {
        get
        {
            if (EffectiveDateTime is not null)
            {
                return EffectiveDateTime;
            }
    
            if (EffectivePeriod is not null)
            {
                return EffectivePeriod;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                EffectiveDateTime = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                EffectivePeriod = (Base.Period)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference? Context { get; set; }
    public string? DateAsserted { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public Base.ResourceReference? MedicationReference { get; set; }
    public Base.ResourceReference[]? PartOf { get; set; }
    public Base.ResourceReference? InformationSource { get; set; }
    public required Base.ResourceReference Subject { get; set; }
    public Base.Period? EffectivePeriod { get; set; }
    public object? Medication    
    {
        get
        {
            if (MedicationCodeableConcept is not null)
            {
                return MedicationCodeableConcept;
            }
    
            if (MedicationReference is not null)
            {
                return MedicationReference;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.CodeableConcept))
            {
                MedicationCodeableConcept = (Base.CodeableConcept)value;
                return;
            }
    
            if (value?.GetType() == typeof(Base.ResourceReference))
            {
                MedicationReference = (Base.ResourceReference)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.ResourceReference[]? ReasonReference { get; set; }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicationStatement.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Period
from base import CodeableConcept
from base import Dosage
from base import DomainResource
from base import Reference
from base import Identifier

@dataclass(kw_only=True)
class MedicationStatement(DomainResource):
    category: Optional[CodeableConcept] = None
    dosage: Optional[List[Dosage]] = field(default_factory=list)
    derived_from: Optional[List[Reference]] = field(default_factory=list)
    reason_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    medication_codeable_concept: Optional[CodeableConcept] = None
    status_reason: Optional[List[CodeableConcept]] = field(default_factory=list)
    note: Optional[List[Annotation]] = field(default_factory=list)
    effective_date_time: Optional[str] = None
    status: str
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    context: Optional[Reference] = None
    date_asserted: Optional[str] = None
    based_on: Optional[List[Reference]] = field(default_factory=list)
    medication_reference: Optional[Reference] = None
    part_of: Optional[List[Reference]] = field(default_factory=list)
    information_source: Optional[Reference] = None
    subject: Reference
    effective_period: Optional[Period] = None
    reason_reference: Optional[List[Reference]] = field(default_factory=list)